src/hooks/__internal__/useModalFocus/useModalFocus.ts

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
```

Comment:
Is it necessary that this logic is a separate hook? It seems like it's not really useful outside of FocusTrap context? Would it be possible to contain it inside of the `focus-trap.component.js` and apply the return values of this hook to the `div ref={trapRef}>{children}</div>` element? If not, maybe you could name it in a way that makes it more descriptive? Like `useModalTabIndex` or `useFocusableModalTabIndex`?

Reply:
The separate hook was simply to avoid repetition, especially as the logic used here is far from obvious. The props returned by the hook have to go in the Dialog (or whatever particular component it is) contained in the Focus Trap rather than FocusTrap itself - but I agree it's worth looking at putting this in FocusTrap and adding the props automatically to the children, thanks for the idea. And thanks for the alternative name suggestions, I like useModalTabIndex.